warehouse/warehouse.go

```diff
@@ -1361,6 +1361,11 @@ func pendingEventsHandler(w http.ResponseWriter, r *http.Request) {
 	// TODO : respond with errors in a common way
 	pkgLogger.LogRequest(r)
 
+	if r.Method != "POST" {
```

Comment:
Instead of using this can we move this to the server mutex?
`mux := mux.NewRouter()`
`mux.HandleFunc("/v1/warehouse/pending-events", pendingEventsHandler).Methods("POST")`

Reply:
We would not need this right.